src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationService.cs

```diff
@@ -238,6 +234,12 @@ internal class CSharpRecommendationService : AbstractRecommendationService
             return symbols;
         }
 
+        private static IEnumerable<ISymbol> GetSymbolsForNamespaceDeclarationNameContext(CSharpSyntaxContext context, CancellationToken cancellationToken)
+        {
+            var declarationSyntax = context.TargetToken.GetAncestor<NamespaceDeclarationSyntax>();
+            return GetRecommendedNamespaceNameSymbols(context.SemanticModel, declarationSyntax, cancellationToken);
```

Comment:
It should only be called when we're sure we're already in an appropriate context. I'll add a check to harden in against misuse.